scylla-proxy/src/proxy.rs

```diff
@@ -942,6 +942,19 @@ mod compression {
             CompressionReader(info),
         )
     }
+
+    fn mock_compression_reader(compression: Option<Compression>) -> CompressionReader {
+        CompressionReader(Arc::new({
+            let once = OnceLock::new();
+            once.set(compression).unwrap();
+            once
+        }))
+    }
+
+    // Compression explicitly turned off.
+    pub(crate) fn no_compression() -> CompressionReader {
+        mock_compression_reader(None)
+    }
```

Comment:
The only uses of this seem to be in tests. Why not move this to tests, or at least mark as cfg(test)?

Reply:
`no_compression()` is used in multiple places out of tests, including `Doorkeeper`.